src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -354,8 +357,29 @@ - (void)startWithWriteable:(id<GRXWriteable>)writeable {
   _retainSelf = self;
 
   _responseWriteable = [[GRXConcurrentWriteable alloc] initWithWriteable:writeable];
+
+  _wrappedCall = [[GRPCWrappedCall alloc] initWithHost:_host path:_path];
+  NSAssert(_wrappedCall, @"Error allocating RPC objects. Low memory?");
+
   [self sendHeaders:_requestHeaders];
   [self invokeCall];
+  // TODO(jcanizales): Extract this logic somewhere common.
+  NSString *host = [NSURL URLWithString:[@"https://" stringByAppendingString:_host]].host;
+  if (!host) {
+    // TODO(jcanizales): Check this on init.
+    [NSException raise:NSInvalidArgumentException format:@"host of %@ is nil", _host];
+  }
+  __weak typeof(self) weakSelf = self;
+  _connectivityMonitor = [GRPCConnectivityMonitor monitorWithHost:host];
```

Comment:
This is interesting:

> From my (adamk) testing on Snow Leopard, 0.0.0.0 seems to be reachable if any network connection is available.

From my tests, it makes a difference if the server is IPv6-only and the device's local networks only supports IPv4: `SCNetworkReachability` will tell you the server isn't reachable. So I rather keep it this way for now. If `SCNetworkReachability` or the underlying `gethostbyname` isn't smart about caching results, and it makes a noticeable impact, we can tune it. And all things said, the real fix for this problem is in the C gRPC library (and will be done there), at which point we can revert this :)